core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -274,41 +273,59 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
         dataSource.inTransaction(_.workspaceQuery.delete(newWorkspace.toWorkspaceName)) >> Future.failed(t)
       }
 
+      jobControlId = cloneResult.getJobReport.getId
+      _ = logger.info(
+        s"Polling on workspace clone in WSM [workspaceId = ${newWorkspace.workspaceIdAsUUID}, jobControlId = ${jobControlId}]"
+      )
+      _ <- traceWithParent("workspaceManagerDAO.getWorkspaceCloneStatus", parentContext) { context =>
+        pollWMCreation(newWorkspace.workspaceIdAsUUID,
+                       jobControlId,
+                       context,
+                       2 seconds,
+                       wsmConfig.pollTimeout,
+                       "Clone workspace",
+                       getWorkspaceCloneStatus
+        ).recoverWith { case t: Throwable =>
```

Comment:
Can you move this block (without the recover) into an inner for-comp with the workspace creation request to that they share the same recovery code? eg
```scala
_ <- (for {
    cloneResult <- traceWithParent("workspaceManagerDAO.cloneWorkspace", parentContext) { ... }
    _ = logger.info( ... )
    _ <- traceWithParent("workspaceManagerDAO.getWorkspaceCloneStatus", parentContext) { ... }
  } yield ()).recoverWith { t: Throwable => ... } 
``

Reply:
(clone job id perhaps isn't necessary to log afterwards)

Reply:
I don't particularly like that because we lose the separate tracing around the 2 WSM calls.

Reply:
Oh wait, I just looked more at your example and see that I can still have the tracing. I'll try it out.

Reply:
Updated.